src/runtime/backend_loader.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ * This file is part of hipSYCL, a SYCL implementation based on CUDA/HIP
+ *
+ * Copyright (c) 2021 Aksel Alpay
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#include "hipSYCL/runtime/backend_loader.hpp"
+#include "hipSYCL/common/debug.hpp"
+#include "hipSYCL/common/config.hpp"
+
+#include <cassert>
+#include <filesystem>
```

Comment:
Maybe use `#include <boost/filesystem.hpp>` instead since we already have a dependency on boost and that way we can support oses that do not have recent enough standard library.

Reply:
`std::filesystem` is part of C++17, which we already require anyway. We don't have a dependency on `boost::filesystem` at the moment. Is there an OS where `std::filesystem` is a problem?

Reply:
Sorry for some reason I thought we require filesystem as well. On Ubuntu 18.04 the standard library is too old. The gcc version is gcc7.5, but filesystem only properly supported in 8.1. Probably the same is true for the centos7. 
https://gcc.gnu.org/onlinedocs/libstdc++/manual/status.html#status.iso.2017

Reply:
Is there a way to update the toolchain? E.g. newer `devtoolset` packages on CentOS? We might want to use other C++17 library features in the future, so giving up on C++17 is not an attractive strategy ;)
Is this even a problem for packaging with spack? Won't it provide its own headers?

Reply:
I don't think spack builds any standard library by default... On centos surely devtoolset can be used, on ubuntu, I have to check. I think the universal solution would be to build and ship our own standard library, we just have to enable `libcxx` when building llvm. (and link against it when building hipSYCL)